framework/test/mocha/unit/modules/chain/forger.js

```diff
@@ -479,18 +592,31 @@ describe('forge', () => {
 
 				forgeModule.config.forging.delegates = [accountDetails];
 
-				forgeModule.loadDelegates(err => {
-					expect(err).to.equal(
-						`Invalid encryptedPassphrase for publicKey: ${
-							accountDetails.publicKey
-						}. Public keys do not match`
-					);
+				return expect(forgeModule.loadDelegates()).to.be.rejectedWith(
+					`Invalid encryptedPassphrase for publicKey: ${
+						accountDetails.publicKey
+					}. Public keys do not match`
+				);
+			});
+
+			it('if publicKeys do not match forgeModule.keypairs should be empty', async () => {
+				const accountDetails = {
+					publicKey:
+						'141b16ac8d5bd150f16b1caa08f689057ca4c4434445e56661831f4e671b7c0a',
+					encryptedPassphrase:
+						'iterations=1&salt=8c79d754416acccb567a42cf62b2e3bb&cipherText=73f5827fcd8eeab475abff71476cbce3b1ecacdeac55a738bb2f0a676d8e543bb92c91e1c1e3ddb6cef07a503f034dc7718e39657218d5a955859c5524be06de5954a5875b4c7b1cd11835e3477f1d04&iv=aac6a3b77c0594552bd9c932&tag=86231fb20e7b263264ca68b3585967ca&version=1',
+				};
+
+				forgeModule.config.forging.delegates = [accountDetails];
+
+				try {
+					await forgeModule.loadDelegates();
```

Comment:
If this call doesn't throw anything this test will give a false positive.
Would be nice to add a falsy expectation after this line to make sure the test never reaches it.

Reply:
Added falsy expectation on all tests using try/catch and testing for failures.